files/dhcp/vrf

```diff
@@ -0,0 +1,178 @@
+get_vrf_arg()
+{
+	local vrf
+
+	vrf=$(echo $1 | tr -cd [:alnum:])
+	if [ "$vrf" != "$1" ]; then
+		echo "Invalid VRF" >&2
+		return 1
+	fi
+
+	echo $vrf
+}
+
+vrf_table()
+{
+	local table_id
+	local vrf
+
+	vrf=$(get_vrf_arg ${1})
+	[ $? -ne 0 ] && return 1
+
+	vrf_exists $vrf
+	if [ $? -eq 0 ]; then
+		vrf_get_table $vrf
+		return 0
+	fi
+
+	# Maybe this is a device, not a vrf.
+	table_id=$(vrf_get_table_dev $vrf)
+	if [ $? -eq 0 ]; then
+		echo ${table_id}
+		return 0
+	fi
+
+	return 1
+}
+
+table_id=$(vrf_table ${interface})
+
+if [ -n "${table_id}" ]; then
+
+case "$reason" in
```

Comment:
where does this script come from, it seems quite different from rfc3442-classless-routes. for example, in rfc3442, it does not have timeout, EXPIRE|FAIL|RELEASE|STOP.

Reply:
Script is port from https://github.com/CumulusNetworks/vrf per our discussion last year.